cpp/src/value_classes/ValueList.cpp

```diff
@@ -225,7 +225,29 @@ bool ValueList::SetByValue
 {
 	// create a temporary copy of this value to be submitted to the Set() call and set its value to the function param
   	ValueList* tempValue = new ValueList( *this );
-	tempValue->m_valueIdx = _value;
+	tempValue->m_valueIdx = GetItemIdxByValue(_value);
```

Comment:
It might be cleaner to also call have ValueList::SetByValue to call SetByIndex below